### PR TITLE
POC sharing at protocol level

### PR DESCRIPTION
**Pull request by gmuraru**
## Description
Move the distribution of the share on a protocol basis because each protocol might have a different version of how it implements sharing. Falcon uses RSS and each party holds 2 shares.

## How has this been tested?
- it should already be tested

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by kamathhrishi** [comment]
I think this abstraction works really well as compared to having different distribute share methods at MPCTensor.

File: src/sympc/protocol/fss/fss.py [hunk 1]
```diff
@@ -180,6 +180,8 @@ def fss_op(x1: MPCTensor, x2: MPCTensor, op="eq") -> MPCTensor:
 class FSS(metaclass=Protocol):
     """Function Secret Sharing."""
```
Comment (kamathhrishi) [inline src/sympc/protocol/fss/fss.py:181]: Add the purpose of the share class to the FSS docstring that describes what it does.

File: src/sympc/protocol/fss/fss.py [hunk 2]
```diff
@@ -180,6 +180,8 @@ def fss_op(x1: MPCTensor, x2: MPCTensor, op="eq") -> MPCTensor:
...
 
+    tensor_class = ShareTensor
```
Comment (kamathhrishi) [inline src/sympc/protocol/fss/fss.py:183]: This should be changed to share_class

Comment (kamathhrishi) [reply src/sympc/protocol/fss/fss.py:183]: Also, why is share_class not an attribute of FSS?

Comment (gmuraru) [reply src/sympc/protocol/fss/fss.py:183]: What do you mean by attribute?

Comment (kamathhrishi) [reply src/sympc/protocol/fss/fss.py]: Ahh, never mind. I got the answer to my question. I just realized protocol functions are static.

**Review by LaRiffle — APPROVED** [review]
Looks good to me!
Any tests need be changed?

**Comment by gmuraru** [comment]
> Looks good to me!
> Any tests need be changed?

I can add a test to test specifcly for this functionality.

**Pull request merged**